Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,427 @@
+lemma interpStrip_eq_of_pos (z : ℂ) (h0 : 0 < sSupNormIm f 0) (h1 : 0 < sSupNormIm f 1) :
+    interpStrip f z = (sSupNormIm f 0) ^ (1 - z) * (sSupNormIm f 1) ^ (z) := by
+  simp only [ne_of_gt h0, ne_of_gt h1, interpStrip, if_false, or_false]
+
+/-- Rewrite for `InterpStrip` when `0 = sSupNormIm f 0` or `0 = sSupNormIm f 1`. -/
+lemma interpStrip_eq_of_zero (z : ℂ) (h : sSupNormIm f 0 = 0 ∨ sSupNormIm f 1 = 0) :
+    interpStrip f z = 0 :=
+  if_pos h
+
+/-- Rewrite for `InterpStrip` on the open vertical strip. -/
+lemma interpStrip_eq_of_mem_verticalStrip (z : ℂ) (hz : z ∈ verticalStrip 0 1):
+    interpStrip f z = (sSupNormIm f 0) ^ (1 - z) * (sSupNormIm f 1) ^ (z) := by
+  by_cases h : ((sSupNormIm f 0) = 0 ∨ (sSupNormIm f 1) = 0)
+  · rw [interpStrip_eq_of_zero _ z h]
+    cases' h with h0 h1
+    · simp only [h0, ofReal_zero, zero_eq_mul, cpow_eq_zero_iff, ne_eq, true_and, ofReal_eq_zero]
+      left
+      rw [sub_eq_zero, eq_comm]
+      simp only [ne_eq, ext_iff, one_re, ne_of_lt hz.2, or_iff_left, false_and, not_false_eq_true]
+    · simp only [h1, ofReal_zero, zero_eq_mul, cpow_eq_zero_iff, ofReal_eq_zero, ne_eq, true_and]
+      right
+      rw [eq_comm]
+      simp only [ne_eq, ext_iff, zero_re, ne_of_lt hz.1, or_iff_left, false_and, not_false_eq_true]
+  · push_neg at h
+    replace h : (0 < sSupNormIm f 0) ∧ (0 < sSupNormIm f 1) := by
+      exact ⟨(lt_of_le_of_ne (sSupNormIm_nonneg f 0) (ne_comm.mp h.1)),
+      (lt_of_le_of_ne (sSupNormIm_nonneg f 1) (ne_comm.mp h.2))⟩
+    exact interpStrip_eq_of_pos f z h.1 h.2
+
+lemma diffContOnCl_interpStrip :
+    DiffContOnCl ℂ (interpStrip f) (verticalStrip 0 1) := by
+  by_cases h : sSupNormIm f 0 = 0 ∨ sSupNormIm f 1 = 0
+  -- Case everywhere 0
+  · eta_expand; simp_rw [interpStrip_eq_of_zero f _ h]; exact diffContOnCl_const
+  -- Case nowhere 0
+  · push_neg at h
+    cases' h with h0 h1
+    rw [ne_comm] at h0 h1
+    apply Differentiable.diffContOnCl
+    intro z
+    eta_expand
+    simp_rw [interpStrip_eq_of_pos f _ (lt_of_le_of_ne (sSupNormIm_nonneg f 0) h0)
+      (lt_of_le_of_ne (sSupNormIm_nonneg f 1) h1)]
+    refine DifferentiableAt.mul ?_ ?_
+    · apply DifferentiableAt.const_cpow (DifferentiableAt.const_sub (differentiableAt_id') 1) _
+      left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+    · refine DifferentiableAt.const_cpow ?_ ?_
+      · apply differentiableAt_id'
+      · left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+
+lemma norm_le_interpStrip_of_mem_verticalClosedStrip_eps (ε : ℝ) (hε : ε > 0) (z : ℂ)
+    (hB : BddAbove ((norm ∘ f) '' (verticalClosedStrip 0 1)))
+    (hd : DiffContOnCl ℂ f (verticalStrip 0 1)) (hz : z ∈ verticalClosedStrip 0 1) :
+    ‖f z‖ ≤  ‖((ε + sSupNormIm f 0) ^ (1-z) * (ε + sSupNormIm f 1) ^ (z) : ℂ)‖ := by
+  simp only [F, abs_invInterpStrip _ _ hε, norm_smul, norm_mul, norm_eq_abs]
+  repeat rw [← ofReal_add]
+  repeat rw [abs_cpow_eq_rpow_re_of_pos (sSupNormIm_eps_pos f hε _) _]
+  simp only [sub_re, one_re]
```

Comment:
```suggestion
  simp only [F, abs_invInterpStrip _ _ hε, norm_smul, norm_mul, norm_eq_abs,
    ← ofReal_add, abs_cpow_eq_rpow_re_of_pos (sSupNormIm_eps_pos f hε _) _, sub_re, one_re]
```
don't hesitate to overuse `simp`, most of the time it's better than `repeat rw`.